packages/components/src/drop-zone/index.js

```diff
@@ -2,25 +2,43 @@
  * External dependencies
  */
 import classnames from 'classnames';
-import { noop } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
+import { Component, createRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Dashicon from '../dashicon';
+import { DropZoneConsumer } from './provider';
 
-class DropZone extends Component {
+const DropZone = ( props ) => (
+	<DropZoneConsumer>
+		{ ( { addDropZone, removeDropZone } ) => (
+			<DropZoneComponent
+				addDropZone={ addDropZone }
+				removeDropZone={ removeDropZone }
+				{ ...props }
+			/>
+		) }
+	</DropZoneConsumer>
+);
+
+class DropZoneComponent extends Component {
 	constructor() {
 		super( ...arguments );
 
-		this.setZoneNode = this.setZoneNode.bind( this );
-
+		this.ref = createRef();
+		this.dropZone = {
```

Comment:
Do we really need this object here? I feel we can just create it in `componentDidMount`

Reply:
Oh never mind, you changed the way we remove the dropzone.